ui/admin/app/components/form/policy/policy-selection/index.js

```diff
@@ -0,0 +1,114 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class FormPolicySelectionComponent extends Component {
+  //methods
+  /**
+   * This is used to show/hide the custom input field
+   * @type {boolean}
+   */
+  get showCustomInput() {
+    if (this.args.name === 'retention_policy') {
+      return this.isCustomRetentionSelected;
+    } else if (
+      this.args.name === 'deletion_policy' &&
+      this.args.model.retain_for?.days >= 0
+    ) {
+      return this.isCustomDeletionSelected;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Returns true if rentention days are greater than 0
+   * @type {boolean}
+   */
+  get isCustomRetentionSelected() {
+    return this.args.model.retain_for?.days > 0;
+  }
+
+  /**
+   * Returns true if deletion days are greater than 0
+   * @type {boolean}
+   */
+  get isCustomDeletionSelected() {
+    return this.args.model.delete_after?.days > 0;
+  }
+  /**
+   * Returns true if the toggle is on
+   * @type {boolean}
+   */
+  get isOverridable() {
+    return this.args.model[this.args.customInputName]?.overridable;
+  }
+
+  get isDeleteDisable() {
+    if (
+      this.args.name === 'deletion_policy' &&
+      this.args.model.retain_for?.days === -1
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  /**
+   * Select options return type is string and we want the `days` in integer format
+   * @type {Number}
+   */
+
+  convertStringToInt(str) {
```

Comment:
I feel like this method doesn't do much and you could have just called `Number(str)` directly